packages/core/src/repository/query/network.query.ts

```diff
@@ -0,0 +1,93 @@
+import { ParameterType } from '../../data';
+import {KeyQuery} from "./query";
+
+export enum HttpMethod {
+    Get = 'GET',
+    Post = 'POST',
+    Put = 'PUT',
+    Delete = 'DELETE',
+}
+
+export abstract class NetworkQuery extends KeyQuery {
+    constructor(key = '') {
+        super(key);
+    }
+
+    abstract get method(): HttpMethod;
+
+    get path(): string {
+        return '/';
+    }
+
+    get body(): unknown | undefined {
+        return undefined;
+    }
+
+    get urlParameters(): Record<string, ParameterType> {
+        return {};
+    }
+
+    get queryParameters(): Record<string, ParameterType> {
+        return {};
+    }
+}
+
+export abstract class NetworkOneQuery extends NetworkQuery {
+    constructor(private readonly pathOne: string, private readonly id: string | number) {
+        super();
+    }
+
+    abstract get method(): HttpMethod;
+
+    public get path(): string {
+        return this.pathOne;
+    }
+
+    public get urlParameters(): Record<string, ParameterType> {
+        return {
+            id: this.id,
+        };
+    }
+}
+
+export class GetOneNetworkQuery extends NetworkOneQuery {
+    public get method(): HttpMethod {
+        return HttpMethod.GET;
+    }
+}
+
+export class PutOneNetworkQuery extends NetworkOneQuery {
```

Comment:
We should also add a `PostOneNetworkQuery`, no?

Reply:
I don't think so, POST requests are made to create the entity so there's no id to complete the url, so it's not a valid use case.